lib/api_hammer/sinatra.rb

```diff
@@ -167,5 +167,25 @@ def parsed_body
         halt_error(415, errors)
       end
     end
+
+    # for methods where parameters which are required in the path may also be specified in the 
+    # body, this takes the path_params and the body object and ensures that they are consistent 
+    # any place they are specified in the body.
+    def check_params_and_object_consistent(path_params, object)
+      errors = {}
+      path_params.each do |(k, v)|
+        if object.key?(k) && object[k] != v
```

Comment:
yeah, the body. this is if you do something like a PUT or PATCH to a resource ( say `/v1/authors/{name}`), such as `PUT /v1/authors/mike {"name": "mike", "age": 99}`, it will check that the `name` in the path is the same as the one in the body, and if not will halt considering it an error.

code I wrote long ago for a personal project and now revive in api_hammer for another project